tools/RAiDER/processWM.py

```diff
@@ -140,16 +136,33 @@ def prepareWeatherModel(
         del weather_model
 
 
-def checkBounds(weather_model, outLats, outLons):
-    '''Check the bounds of a weather model'''
-    ds = xr.load_dataset(weather_model.files[0])
-    coords = ds.coords  # coords is dict-like
-    keys = [k for k in coords.keys()]
-    xc = coords[keys[0]]
-    yc = coords[keys[1]]
-    lat_bounds = [yc.min(), yc.max()]
-    lon_bounds = [xc.min(), xc.max()]
-    self_extent = lat_bounds + lon_bounds
-    in_extent = weather_model._getExtent(outLats, outLons)
+def checkContainment(weather_model: weatherModel,
+                     outLats: np.ndarray,
+                     outLons: np.ndarray) -> bool:
+    '''Check if weather model contains bounding box of outLats and outLons'''
+    weather_model_path = weather_model.files[0]
+    with rasterio.open(f'netcdf:{weather_model_path}') as ds:
+        datasets = ds.subdatasets
+
+    with rasterio.open(datasets[0]) as ds:
+        bounds = ds.bounds
```

Comment:
@dbekaert - I assume the netcdf has a bunch of arrays of similar shape as in they conform to the xarray dataset. I can see this would fail for the ARIA-type netcdfs if there are groups with different shapes/coordinates as well as a shapefile embedded there.

Reply:
I guess, you are assuming that any first dataset hit would be the right size.

Reply:
Yes, exactly. If you would like a different approach, fine to do that. Let me know.

Reply:
Depending how we would go about xarray in future, we could capture this dataset and variable mapping.